examples/crm/src/SignupPage.tsx

```diff
@@ -0,0 +1,126 @@
+import { Button, Container, Stack, TextField, Typography } from '@mui/material';
+import { useCreate, useGetList, useLogin, useNotify } from 'react-admin';
+import { useForm, SubmitHandler } from 'react-hook-form';
+import { Navigate } from 'react-router';
+
+interface UserInput {
+    full_name: string;
+    email: string;
+    password: string;
+}
+
+export const SignupPage = () => {
+    const { total } = useGetList('users', {
+        pagination: { page: 1, perPage: 10 },
+        sort: { field: 'name', order: 'ASC' },
+    });
+
+    const login = useLogin();
+    const notify = useNotify();
+    const [create] = useCreate();
+
+    const {
+        register,
+        handleSubmit,
+        formState: { isValid },
+    } = useForm<UserInput>({
+        mode: 'onChange',
+    });
+
+    // For the moment, we only allow one user to sign up. Other users must be created by the administrator.
```

Comment:
Maybe you should also handle the `isPending` state here, to avoid showing the form if response takes ages to be retrieved ?